### PR TITLE
bring C++ and Java APIs into sync

### PR DESCRIPTION
**Pull request by arnej27959**
* detect and make accessible port numbers to use for
  config server / config proxy from Java also.
* port numbers are relative to VESPA_PORT_BASE if not
  explicitly specified.
* try to make the logic in C++ and Java more similar.

@hmusum please review and merge

### CODE REVIEW

File: defaults/src/main/java/com/yahoo/vespa/defaults/Defaults.java [hunk 1]
```diff
@@ -27,15 +27,20 @@
...
         vespaHome = findVespaHome();
         vespaUser = findVespaUser();
         vespaHost = findVespaHostname();
-        vespaWebServicePort = findVespaWebServicePort();
-        vespaPortBase = 19000; // TODO
+        vespaWebServicePort = findWebServicePort(8080);
+        vespaPortBase = findVespaPortBase(19000);
+        vespaPortConfigServerRpc = findConfigServerPort(vespaPortBase + 70);
+        vespaPortConfigServerHttp = vespaPortConfigServerRpc + 1;
+        vespaPortConfigProxyRpc = findConfigProxyPort(vespaPortBase + 91);
```
Comment (hmusum) [inline defaults/src/main/java/com/yahoo/vespa/defaults/Defaults.java]: I think this should be 90

**Review by hmusum — CHANGES_REQUESTED** [review]

File: defaults/src/main/java/com/yahoo/vespa/defaults/Defaults.java (hunk 1 above)
Comment (arnej27959) [reply defaults/src/main/java/com/yahoo/vespa/defaults/Defaults.java]: right, no idea how i ended up with 91 here...

File: defaults/src/apps/printdefault/printdefault.cpp [hunk 2]
```diff
@@ -15,6 +39,8 @@ int main(int argc, char **argv) {
     }
     if (strcmp(argv[1], "home") == 0) {
         printf("%s\n", vespa::Defaults::vespaHome());
+    } else if (strcmp(argv[1], "everything") == 0) {
```
Comment (hmusum) [inline defaults/src/apps/printdefault/printdefault.cpp]: Nitpick: I am more used to "all" than "everything"

**Review by hmusum — APPROVED** [review]
One minor comment, merge at will

**Pull request merged**